scalafix-core/src/main/scala/scalafix/rewrite/NoXml.scala

```diff
@@ -0,0 +1,42 @@
+package scalafix
+package rewrite
+
+import scala.meta._
+import patch.TreePatch.AddGlobalImport
+
+case object NoXml extends Rewrite {
+
+  override def rewrite(ctx: RewriteCtx): Patch = {
+    val patch = ctx.tree.tokens.collect {
+      case tok @ Token.Xml.Start() =>
+        ctx.addLeft(tok, "xml\"\"\"")
+
+      case tok @ Token.Xml.End() =>
+        ctx.addRight(tok, "\"\"\"")
+
+      case tok @ Token.Xml.SpliceStart() =>
+        ctx.addLeft(tok, "$")
+
+      case tok @ Token.Xml.Part(part) =>
+        if (part.contains('$'))
+          ctx.replaceToken(tok, part.replaceAllLiterally("$", "$$"))
+        else Patch.empty
+
+    }.asPatch
+
+    // Emit warning on `{{`
```

Comment:
Good question. I've actually considered replacing this mutable reporter (`info(): Unit`) with empty patches like `ctx.info(msg, pos): Patch`. That should make it easier to test.